staff/jaume-serradell/spotify-app/scripts/main.js

```diff
@@ -0,0 +1,94 @@
+// my presentation logic
+
+// optional, reduce the size of the restaurants loaded in memory
+// restaurants.splice(100);
+
+var search = new SearchPanel();
+
+search.onSearch(function (query) {
+    logic.token = 'BQDvjaDo0bwTjgm_AuMTpolUiJXNApTsQ0qKK5HK9jcbYqn9MvAn1gMwlomsvC-th22arnA2LTgLaCL-9UlhGOJUD_5aSRZXDwrRPPRrtg1HrlXaNWN3BTwoL_cBlzcYXAc2VOSdz0FgsLFGZe8-JY4_5zZRlnf3Uz6LvBp_tAOzzwa6qq8z8IIflnnIHA_iAjN0lDOC9aC2WZ5nxC_bdV3aSjhkkbrq0D6whgvjcizWYdVMBeXX1rCiC3G-bXHunaaABhEmA40';
+    logic.searchArtists(query)
+        .then(function (artists) {
+            results.updateResults(artists.map(function (artist) {
+                return {
+                    id: artist.id,
+                    text: artist.name
+                };
+            }));
+
+            detailContainer.clear();
+        })
+        .catch(function (error) {
+            alert('Sorry, we have temporary problem, try again later.');
+        });
+});
+
+var results = new ResultsList();
```

Comment:
You should overwrite the updateResults() function in order to include more information.